C-18-01-Capstone.Main/DataAccessLayer/DataAccessLogger.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C_18_01_Capstone.Main.DataAccessLayer
+{
+   public class DataAccessLogger<T> : IDataAccess<T> where T : class, new()
+   {
+      private IDataAccess<T> entity;
+      private string path = "DataServiceLog.txt";
+
+
+      public DataAccessLogger(IDataAccess<T> entity)
+      {
+         this.entity = entity;
+      }
+
+      public void AddEntity(T entity)
+      {
+         string logString = string.Format("{0}({1})", nameof(this.AddEntity), typeof(T).Name);
+         LogDataStart(logString);
+         this.entity.AddEntity(entity);
+         LogDataEnd(logString);
+      }
+
+      public void DeleteEntity(T entity)
+      {
+         string logString = string.Format("{0}({1})", nameof(this.DeleteEntity), typeof(T).Name);
+         LogDataStart(logString);
+         this.entity.DeleteEntity(entity);
+         LogDataEnd(logString);
+      }
+
+      private void LogDataStart(string proccessName)
+      {
+         string log = LogString("Starting", proccessName);
+
+         File.AppendAllText(path,
```

Comment:
replace with filestream (text writer). use Writeline method from it and get rid of newlines